### PR TITLE
Update handling of missing participants

### PR DESCRIPTION
**Pull request by typfel**
## What's new in this PR?

Add method for verifying that a user is part of conversation.

## Note

Deleted some code for moving security level messages which is obsolete since we support adding/removing participants locally anymore.

### CODE REVIEW

File: Source/Model/Conversation/ZMConversation+SecurityLevel.swift [hunk 1]
```diff
@@ -142,6 +141,33 @@ extension ZMConversation {
...
+    /// Adds the user to the list of participants if not already present and inserts a .participantsAdded system message
+    @objc(addParticipantIfMissing:date:)
+    public func addParticipantIfMissing(_ user: ZMUser, at date: Date = Date()) {
+        guard !activeParticipants.contains(user) else { return }
+        
+        switch conversationType {
+        case .group:
+            appendSystemMessage(type: .participantsAdded, sender: user, users: Set(arrayLiteral: user), clients: nil, timestamp: date)
+            internalAddParticipants(Set(arrayLiteral: user))
+            break
```
Comment (vytis) [inline Source/Model/Conversation/ZMConversation+SecurityLevel.swift]: No need for extra breaks here

File: Source/Model/Conversation/ZMConversation.m [hunk 2]
```diff
@@ -1530,11 +1471,11 @@ - (NSSet *)messagesNotToRefreshBecauseNeededForSorting
         NSUInteger currentIndex = length-1;
         const NSUInteger keepUntilIndex = (length-1 >= NumberOfMessagesToKeep) // avoid overflow
         ? (length-1 - NumberOfMessagesToKeep)
-        : length-1;
+        : 0;
```
Comment (vytis) [inline Source/Model/Conversation/ZMConversation.m]: Maybe it's worth rewriting this method to something more readable?

File: Tests/Source/Model/Conversation/ZMConversationTests+SecurityLevel.m [hunk 3]
```diff
@@ -711,36 +722,52 @@ - (void)testItResendsAllMessagesThatCausedDegradation
...
         XCTAssertTrue(message3.isExpired);
-        XCTAssertTrue(message3.causedSecurityLevelDegradation);
+        XCTAssertFalse(message3.causedSecurityLevelDegradation);
+        XCTAssertEqual(message3.deliveryState, ZMDeliveryStateFailedToSend);
+        XCTAssertFalse(conversation.allUsersTrusted);
+        XCTAssertEqual(conversation.securityLevel, ZMConversationSecurityLevelSecureWithIgnored);
+    }];
+}
+
+- (void)testIMarksConversationAsNotSecureAfterResendMessages
```
Comment (vytis) [inline Tests/Source/Model/Conversation/ZMConversationTests+SecurityLevel.m]: typo - testIMarks

**Comment by typfel** [comment]
@zenkins release minor

**Pull request merged**